### PR TITLE
Add graphql-scalar blog post draft

### PR DESCRIPTION
**Pull request by eddeee888**
This PR adds a new blog post.

TODOs:

- [x] Title
- [x] Description
- [x] Image
- [x] Thumbnail
- [x] Image TODO: `Add the image showing the scalar flow related to request here`
- [x] Callouts to the recent scalar change
- [x] Final draft

### CODE REVIEW

**Comment by Urigo** [comment]
finally everything scalars in one place!

**Review by n1ru4l — COMMENTED** [review]
Maybe it is worth mentioning previous pain points, one that I have in mind and encountered a lot is that within a graphql resolver, I wanted to return an object that had a `number` id property. Because of that I explicitly had to either declare a mapper type or rebuild the whole object returned from the resolver just for stringifying the id in order to please TypeScript

**Comment by eddeee888** [comment]
> Maybe it is worth mentioning previous pain points, one that I have in mind and encountered a lot is that within a graphql resolver, I wanted to return an object that had a `number` id property. Because of that I explicitly had to either declare a mapper type or rebuild the whole object returned from the resolver just for stringifying the id in order to please TypeScript

Great callout! I've added it here: https://github.com/the-guild-org/website/pull/1279/files#diff-b95dbe6e7ed8bc6f7e3b37205c6cc41974ca1620200dcc4f11ea0e0e694ef972R296

File: website/pages/blog/complete-guide-to-graphql-scalar.mdx [hunk 1]
```diff
@@ -0,0 +1,423 @@
+---
+title: Complete Guide to GraphQL Scalar
```
Comment (eddeee888) [inline website/pages/blog/complete-guide-to-graphql-scalar.mdx:2]: I'm still trying to decide whether to include the following:

- Link to https://scalars.graphql.org/ explaining what it's for
- (Lack of) Client type for `graphql-scalars`

File: website/pages/blog/complete-guide-to-graphql-scalar.mdx [hunk 2]
```diff
@@ -0,0 +1,423 @@
...
+  generates: {
+    'src/schema/types.generated.ts': {
+      plugins: ['typescript', 'typescript-resolvers'],
+      config: {
+        scalars: {
+          ID: {
+            input: 'string',
+            output: 'string | number',
+          },
+          DateTime: DateTimeResolver.extensions.codegenScalarType // 2. Use scalar resolver's recommended server type
```
Comment (eddeee888) [inline website/pages/blog/complete-guide-to-graphql-scalar.mdx:368]: Is this the right way of using the extensions type? 🤔 

There's something in the implementation here but I'm not 100% sure how it works: https://github.com/dotansimha/graphql-code-generator/blob/9e8bd9d1039646217ef0e6007e3a349631bdd1a3/packages/plugins/other/visitor-plugin-common/src/utils.ts#L358-L367

File: website/pages/blog/complete-guide-to-graphql-scalar.mdx [hunk 3]
```diff
@@ -0,0 +1,423 @@
+---
+title: Complete Guide to GraphQL Scalar
+authors: eddeee888
+tags: [graphql, scalars, codegen, node, server, typescript]
+date: 2023-06-19
+description: Knowing how native and custom GraphQL Scalar works enables building flexible and extendable GraphQL schema. 
+image: /blog-assets/scalable-apis-with-graphql-server-codegen-preset/cover.png
+thumbnail: /blog-assets/scalable-apis-with-graphql-server-codegen-preset/thumbnail.png
```
Comment (eddeee888) [inline website/pages/blog/complete-guide-to-graphql-scalar.mdx:8]: I'll need help with the image and thumbnail 🙏

**Review by n1ru4l — APPROVED** [review]

**Pull request merged**